### PR TITLE
Azure Service Bus support

### PR DESCRIPTION
**Pull request by baitcode**
Implemented Azure broker
Updated README a bit
Cleaned up gradle warnings:
  * Pinned java target to Java 18
  * Updated kotlin and jvm to the latest
Changed tests to use until to wait till condition and the assert, increased time to wait due to poor Azure performance
Added logging configuration to tests too much logs otherwise

### CODE REVIEW

File: src/main/kotlin/AzureServiceBusBroker.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+    override fun submitMessage(queueName: QueueName, message: Message) {
+        assertQueue(queueName)
+
+        val msg = ServiceBusMessage(message.body)
+            .setMessageId(UUID.randomUUID().toString())
+
+        val kotaskHeaders = message.headers
+            .mapKeys { (k, _) -> "$HEADERS_PREFIX$k" }
+            .plus(delayHeaderName to message.delayMs.toString())
```
Comment (ilyatikhonov) [inline src/main/kotlin/AzureServiceBusBroker.kt:41]: am I right that this delay header is not necessary for the Service Broker delay mechanism? Maybe we can omit it?

File: src/main/kotlin/AzureServiceBusBroker.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    .toMap()
+
+                val msg = Message(
+                    message.body.toBytes(),
+                    headers = headers,
+                    delayMs = allHeaders[delayHeaderName]?.toString()?.toLong() ?: 0,
+                )
+
+                runBlocking {
+                    handler(msg) { }
```
Comment (ilyatikhonov) [inline src/main/kotlin/AzureServiceBusBroker.kt:91]: message Ack should be done in the handlers callback:

```
typealias ConsumerHandler = suspend (message: Message, ack: ()->Any) -> Unit
```

**Review by ilyatikhonov — CHANGES_REQUESTED** [review]
Good job! Take a look at the Ack comment

File: src/main/kotlin/AzureServiceBusBroker.kt (hunk 1 above)
Comment (baitcode) [reply src/main/kotlin/AzureServiceBusBroker.kt:41]: I decided to keep to keep messages in different brokers look as similar as possible.

File: src/main/kotlin/AzureServiceBusBroker.kt (hunk 2 above)
Comment (baitcode) [reply src/main/kotlin/AzureServiceBusBroker.kt]: It took me a while to decide how to write this. But yeah, you are right. Architecture-wise this is better solution.

**Comment by ilyatikhonov** [comment]
Lgtm

**Pull request merged**